src/data.rs

```diff
@@ -1427,6 +1433,16 @@ unsafe impl<'a> Binding<'a> for DataNodeRef<'a> {
     ) -> DataNodeRef<'a> {
         DataNodeRef { tree, raw }
     }
+
+    unsafe fn into_raw(self) -> *mut Self::CType {
+        let raw = self.raw;
+        #[allow(clippy::forget_non_drop)]
+        // Allow the forget here despite DataNodeRef not having a Drop impl. If
+        // a Drop impl ever is added, we don't want this to start behaving
+        // incorrectly.
+        std::mem::forget(self);
+        raw
```

Comment:
I think the following code would be equivalent but simpler:
```rs
let this = ManuallyDrop::new(self);
this.raw
```

Reply:
Definitely simpler. I always forget they added `ManuallyDrop`.